src/h5web/visualizations/index.test.ts

```diff
@@ -1,559 +1,408 @@
-        },
-      ];
-
-      unsupportedDatasets.forEach((dataset) => {
-        expect(supportsEntity(Vis.Heatmap, dataset)).toBe(false);
-      });
+    it('should not support dataset with non-numeric type', () => {
+      expect(supportsEntity(datasetStr2D)).toBe(false);
     });
 
-    it('should not support group and datatype', () => {
-      const unsupportedEntities: HDF5Entity[] = [
-        { ...baseGroup },
-        { ...baseDatatype, type: '' },
-      ];
+    it('should not support dataset with non-simple shape', () => {
+      expect(supportsEntity(datasetIntScalar)).toBe(false);
+    });
 
-      unsupportedEntities.forEach((entity) => {
-        expect(supportsEntity(Vis.Heatmap, entity)).toBe(false);
-      });
+    it('should not support dataset with less than two dimensions', () => {
+      expect(supportsEntity(datasetInt1D)).toBe(false);
+    });
+
+    it('should not support group and datatype', () => {
+      expect(supportsEntity(groupEmpty)).toBe(false);
+      expect(supportsEntity(datatypeAdvanced)).toBe(false);
     });
   });
 
   describe('Vis.NxSpectrum', () => {
-    const nxSpectrumDef = VIS_DEFS[Vis.NxSpectrum];
+    const supportsEntity = makeSupportFn(Vis.NxSpectrum);
 
     it('should support NXdata group referencing signal with spectrum interpretation', () => {
-      const metadata = {
-        datasets: {
-          [baseDataset.id]: {
-            ...baseDataset,
-            attributes: [
-              { name: 'interpretation', value: NxInterpretation.Spectrum },
-            ],
-          },
-        },
-      } as HDF5Metadata;
-
-      const supported = nxSpectrumDef.supportsEntity(nxDataGroup, metadata);
-      expect(supported).toBe(true);
-    });
-
-    it('should support NXdata group referencing signal with numeric type, simple shape, one dimension and no or unknown interpretation', () => {
-      const supportedSignals: HDF5Entity[] = [
-        {
-          ...baseDataset,
-          type: { class: HDF5TypeClass.Integer, base: 'H5T_STD_I32LE' },
-          shape: { class: HDF5ShapeClass.Simple, dims: [1] },
-          attributes: [], // no interpretation
-        },
-        {
-          ...baseDataset,
-          type: { class: HDF5TypeClass.Float, base: 'H5T_IEEE_F32BE' },
-          shape: { class: HDF5ShapeClass.Simple, dims: [1] },
-          attributes: [
-            { name: 'interpretation', value: 'unknown' } as HDF5Attribute, // unknown interpretation
-          ],
-        },
-      ];
-
-      supportedSignals.forEach((signal) => {
-        const metadata = {
-          datasets: { [baseDataset.id]: signal },
-        } as HDF5Metadata;
-
-        const supported = nxSpectrumDef.supportsEntity(nxDataGroup, metadata);
-        expect(supported).toBe(true);
+      const group = makeNxDataGroup('foo', {
+        signal: 'my_signal',
+        ids: { my_signal: datasetIntScalar.id }, // dataset type and shape doesn't matter
```

Comment:
Yep... and our philosophy so far for malformed NeXuS has been to avoid errors, and will be to allow users to debug them somehow as per #254. So yeah, we should add a check in `supportsEntity`. I'll think about it.

Reply:
Ok, so I've updated the logic in `supportsEntity` and I've added an extra test for each Nx vis. So we now have two tests with the correct interpretation: one with a dataset with a supported type and shape and one with an unsupported dataset.